src/pretix/plugins/banktransfer/views.py

```diff
@@ -886,3 +891,23 @@ def get_object(self, *args, **kwargs):
             organizer=self.request.organizer,
             pk=self.kwargs.get('id')
         )
+
+
+@method_decorator(xframe_options_exempt, 'dispatch')
+class SendInvoiceMailView(EventViewMixin, OrderDetailMixin, View):
+    def post(self, request, *args, **kwargs):
+        if not self.order:
+            raise Http404(_('Unknown order code or not authorized to access this order.'))
+
+        last_payment = self.order.payments.last()
+        last_invoice = self.order.invoices.last()
+        if not last_payment or not last_invoice:
+            messages.error(request, _('No payment or invoice found, please request an invoice first.'))
+            return redirect(self.get_order_url())
+        provider = last_payment.payment_provider
+        provider.send_invoice_to_alternate_email(self.order, last_invoice, request.POST['email'])
```

Comment:
There is no validation going on here that the email address is in fact a valid email address. We should at least use `EmailValidator()` from Django, or (maybe even more "clean", maybe overkill, depending on taste) use a django form class.

Reply:
I added a validation using validate_email. 

I'd say a Django Form is not necessary as it is only one field and it is also validated by type=email client-side, so a low probability that a user will experience a server-side validation error. 

(Also, I'm not sure how a Django Form could be implemented here, because form and validation are in different views. I guess we could store the validation errors in the session, redirect back to the order view, and display them there. I think this would also unnecessarily complicate the payment_pending_render method, because we'd need to instantiate the Form there as well.)

Reply:
> I'd say a Django Form is not necessary as it is only one field and it is also validated by type=email client-side, so a low probability that a user will experience a server-side validation error.

I'm totally okay with that.

> (Also, I'm not sure how a Django Form could be implemented here, because form and validation are in different views. I guess we could store the validation errors in the session, redirect back to the order view, and display them there? However, I think this would unnecessarily complicate the payment_pending_render method, because we'd need to instantiate the Form there as well.)

Uh yeah, that's maybe too complicated, yes.